sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -102,73 +138,231 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
         .innerJoin("offering.institutionLocation", "institutionLocation")
         .getOne();
 
-      // Cloning existing offering.
-      const offering: EducationProgramOffering = { ...existingOffering };
-
-      // Assigning id as undefined, so that when its saved its considered as a new EducationProgramOffering object.
-      offering.id = undefined;
-
-      const newStudyEndDate =
-        scholasticStandingData.dateOfChange ??
-        scholasticStandingData.dateOfCompletion ??
-        scholasticStandingData.dateOfIncompletion ??
-        scholasticStandingData.dateOfWithdrawal;
-
-      offering.studyEndDate = new Date(newStudyEndDate);
-
-      offering.actualTuitionCosts =
-        scholasticStandingData.tuition ?? existingOffering.actualTuitionCosts;
-      offering.programRelatedCosts =
-        scholasticStandingData.booksAndSupplies ??
-        existingOffering.programRelatedCosts;
-      offering.mandatoryFees =
-        scholasticStandingData.mandatoryFees ?? existingOffering.mandatoryFees;
-      offering.exceptionalExpenses =
-        scholasticStandingData.exceptionalCosts ??
-        existingOffering.exceptionalExpenses;
-
-      offering.offeringType = OfferingTypes.ScholasticStanding;
+      // Check for restrictions and apply if any.
+      const studentRestriction = await this.manageStudentRestrictions(
```

Comment:
Is there a scenario where no restriction will be needed? Maybe because the ones needed were already added?

Reply:
for the first two options, there is no restriction 
![image](https://user-images.githubusercontent.com/77353155/172723152-029aab3b-aad6-4270-92d2-770ffb0051aa.png)

Reply:
and for the 3rd option if the sum of unsuccessful weeks is less than 68, then also no restriction

Reply:
That is what I am saying, I did not see a check if the method `this.manageStudentRestrictions` returns `null`.

Reply:
it will return  null for the first 2 options and for the 3rd option when week less than 68

Reply:
I saw that the check is now in the code, thanks 😉